tests/e2e/network/service_annotations.go

```diff
@@ -315,3 +346,57 @@ func validateInternalLoadBalancer(c clientset.Interface, ns string, url string)
 	utils.Logf("validation finished")
 	return err
 }
+
+func validateLoadBalancerBackendPools(vmssname string, cs clientset.Interface, labels map[string]string, ns string, ports []v1.ServicePort, resourcegrpverify string) {
+	// create deployment
+	serviceName := "validate-lb-mode"
+	lbModeLoadBalancerName := serviceName + string(uuid.NewUUID())
+
+	utils.Logf("Creating deployment " + lbModeLoadBalancerName)
+	deployment := createNginxDeploymentManifest(lbModeLoadBalancerName, labels)
+
+	// create service annotation mode for LoadBalancer
+	mode_name := vmssname
+	annotation := map[string]string{
+		azure.ServiceAnnotationLoadBalancerMode: mode_name,
+	}
+	service := createLoadBalancerServiceManifest(cs, lbModeLoadBalancerName, annotation, labels, ns, ports)
+	_, er := cs.CoreV1().Services(ns).Create(service)
+	Expect(er).NotTo(HaveOccurred())
+	utils.Logf("Successfully created LoadBalancer service " + lbModeLoadBalancerName + " in namespace " + ns)
+
+	By("Waiting for service exposure")
+	pip, err := utils.WaitServiceExposure(cs, ns, serviceName)
+	Expect(err).NotTo(HaveOccurred())
+	
+	iplist, er := utils.ListPublicIPs(cs, resourcegrpverify)
+	Expect(er).NotTo(HaveOccurred())
+	for _, ip := range iplist {
+		if ip.Key == pip {
```

Comment:
please use `ip.PublicIPAddressPropertiesFormat.IPAddress`  instead of ip.Key. Note the properties PublicIPAddressPropertiesFormat and PublicIPAddressPropertiesFormat.IPAddress are pointers, so you need check whether they are nil

Reply:
I think it must be `PublicIPAddressPropertiesFormat.ipAddress`